scripts/export_sqlite_to_workspace.py

```diff
@@ -0,0 +1,259 @@
+                sett[k] = dv # copy default port no
+            else:
+                sett[parts[0] + "_enabled"] = True
+                sett[k] = int(csett[k])
+        elif k == "pstorage_polling":
+            if csett.get(k) == "disabled":
+                sett[parts[0] + "_enabled"] = False
+                sett[k] = dv  # copy default port no
+            else:
+                sett["pstorage_enabled"] = True
+                sett[k] = int(cv)
+        elif k == "start_run_mode":
+            sett["start_run_mode"] = bool(cv)
+        elif parts[1] == "enabled":
+            sett[k] = bool(cv)
+        else:
+            sett[k] = int(cv)
+
+    write_json(None, "settings.json", sett)
+    write_json(None, "settings.default.json", defSett)
+
+    etc_dir = os.path.abspath(os.path.join(openplc, "etc"))
+    for fn in "active_program", "openplc_driver","openplc_platform":
+        fp = os.path.join(etc_dir, fn)
+        contents = ""
+        with open(fp, "r") as f:
+            contents = f.read()
+            f.close()
+        sf = os.path.join(workspace, "%s.txt" % fn)
+        with open(sf, "w") as f:
+            f.write(contents.strip())
+            f.close()
+            print("Wrote: %s" % sf)
+
+    ## uploads demo
+    up = os.path.join(workspace, "uploads")
+    if not os.path.exists(up):
+        os.mkdir(up)
+
+    st = os.path.join(openplc, "etc", "st_files", "blank_program.st")
+    tar = os.path.join(up, "%s.blank_program.st" % gen_uid())
+    shutil.copyfile(st, tar)
+    print("Copied: %s" % tar)
+
+    ## archive
+    conn.close()
+
+    ark = os.path.join(workspace, "archive")
+    if not os.path.exists(ark):
+        os.mkdir(ark)
+    fn = "%s.%s" % (gen_uid(), os.path.basename(database))
+    dbak = os.path.join(ark, fn)
+    shutil.copyfile(database, dbak)
+    print("Copied: %s" % dbak)
+
+
+
+    print("DONE :-)")
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Export sqlite to workspace folder")
+
+    parser.add_argument("-w", "--workspace",
+                        help="Path to work directory",
+                        action="store", type=str)
+
+    parser.add_argument("-d", "--database",
+                        help="Path to sqlite db",
+                        action="store", type=str)
+
+    parser.add_argument("-o", "--openplc",
+                        help="Path to openplc checkout",
+                        action="store", type=str)
+
+    args = parser.parse_args()
+
+    if args.database == None:
+        print("ERROR: need a -d database")
+        parser.print_help()
+        sys.exit(0)
+
+    if args.workspace == None:
+        print("ERROR: need a -w work directory")
+        parser.print_help()
+        sys.exit(0)
+
+    if args.openplc == None:
+        print("ERROR: Need -o for openplc install")
+        parser.print_help()
+        sys.exit(0)
+
+    export_db(database=args.database, workspace=args.workspace, openplc=args.openplc)
```

Comment:
This should set the exit code to 0 if successful, or non-zero not successful. That means the function export_db should return an integer value